arangod/Agency/Agent.cpp

```diff
@@ -515,8 +531,8 @@ priv_rpc_ret_t Agent::recvAppendEntriesRPC(term_t term, std::string const& leade
   {
     WRITE_LOCKER(oLocker, _outputLock);
     CONDITION_LOCKER(guard, _waitForCV);
-    index_t const tmp = std::max(_commitIndex, std::min(leaderCommitIndex, lastIndex));
-    if (tmp > _commitIndex) {
+    index_t const tmp = std::max(_commitIndex.load(std::memory_order_relaxed), std::min(leaderCommitIndex, lastIndex));
```

Comment:
`_commitIndex` is loaded twice in a row here.
If it doesn't have additional synchronization, then both loads could produce different values.
Even if not, it will be more efficient to load the value just once, and use it in both expressions.